shopfloor_single_product_transfer/services/single_product_transfer.py

```diff
@@ -521,21 +616,52 @@ def _write_destination_on_lines(self, lines, location):
             self._lock_lines(checkout_sync._all_lines_to_lock(lines))
             checkout_sync._sync_checkout(lines, location)
         lines.location_dest_id = location
-        lines.package_level_id.location_dest_id = location
 
-    def _set_quantity__post_move(self, location, move_line, confirmation=False):
+    def _set_quantity__post_move(self, move_line, location, confirmation=False):
         # TODO qty_done = 0: transfer_no_qty_done
         # TODO qty done < product_qty: transfer_confirm_done
         self._write_destination_on_lines(move_line, location)
-        stock = self._actions_for("stock")
-        stock.validate_moves(move_line.move_id)
+        self._post_move(move_line)
         message = self.msg_store.transfer_done_success(move_line.picking_id)
         completion_info = self._actions_for("completion.info")
         completion_info_popup = completion_info.popup(move_line)
         return self._response_for_select_product(
-            move_line.location_id, message=message, popup=completion_info_popup
+            location=move_line.location_id, message=message, popup=completion_info_popup
         )
 
+    def _post_move(self, move_line):
+        # TODO: when we split the move, we still get a
+        # backorder, which should not be the case.
+        # See if there's a way to identify the moves
+        # generated through this mechanism and avoid creating them.
+        move_line._split_partial_quantity()
+        new_move = move_line.move_id.split_other_move_lines(
+            move_line, intersection=True
+        )
+        new_move.extract_and_action_done()
```

Comment:
```suggestion
        # if not all the quantity has been processed, split the move line
        move_line._split_partial_quantity()
        # and then create a new move for the processed move line
        new_move = move_line.move_id.split_other_move_lines(
            move_line, intersection=True
        )
       # Validate the new created move (in case of partial quantity) or the initial move (in case of full quantity) 
       (new_move or move_line.move_id).extract_and_action_done()                                  
```